Creating scenarios for RAID 1,5,6,10

hjluo opened the pull request:
##
### Creating scenarios for RAID 1,5,6,10

- Related ticket: https://progress.opensuse.org/issues/184177
- Needles: N/A
- Verification run: 
   - TBA

##

jknphy commented:
having now power 10 kvm, I don't see any reason to not have even the RAID 0 also included, could you address that, please.

jknphy commented on data/yam/agama/auto/lib/storage.libsonnet:
```diff
@@ -188,6 +188,14 @@ local raid(level='raid0', uefi=false) = {
+  raid6: raid('raid6'),
+  raid6_uefi: raid('raid6', true),
+  raid10: raid('raid10'),
+  raid10_uefi: raid('raid10', true),
```
we don't need uefi for all of them, just raid0 (and that is not for raid testing, but for the other feature to edit existing raid), in general having some minimal test coverage of uefi is enough.

okurz reviewed (changes requested):
The commit message could have a bit more details, e.g. what issue is fixed, why this needs to change, to which versions of which product it applies, link to a bug or ticket entry, the choices you made, etc. Also see https://commit.style/ or http://chris.beams.io/posts/git-commit/ as a helpful guide how to write good commit messages.

Also your message says "Creating scenarios for RAID…" as if no such scenarios would have existed but RAID is covered in openQA tests since 10+ years

jknphy commented on data/yam/agama/auto/lib/storage.libsonnet:
```diff
@@ -136,6 +136,17 @@ local raid(level='raid0', uefi=false) = {
       ],
     },
+  ] else if boot_type == 'prep' then [
+    // PowerPC PReP boot partition case
```
it is well known that for other architectures there are special boot partitions, you can omit the comment.

jknphy commented on data/yam/agama/auto/lib/storage.libsonnet:
```diff
@@ -136,6 +136,17 @@ local raid(level='raid0', uefi=false) = {
+    {
+      search: '*',
+      partitions: [
+        { delete: true, search: '*' },
```
why are you deleting partitions if we have fresh disk in qemu for the 3 architectures? shouldn't be needed.

jknphy commented:
LGTM code-wise other than addressing all those comments above.

hjluo commented:
> The commit message could have a bit more details, e.g. what issue is fixed, why this needs to change, to which versions of which product it applies, link to a bug or ticket entry, the choices you made, etc. Also see [commit.style](https://commit.style/) or [http://chris.beams.io/posts/git-commit](http://chris.beams.io/posts/git-commit/) as a helpful guide how to write good commit messages.
> 
> Also your message says "Creating scenarios for RAID…" as if no such scenarios would have existed but RAID is covered in openQA tests since 10+ years

Addressed your comments, please take a look again.Thanks.

lemon-suse commented on tests/console/validate_raid.pm:
```diff
@@ -73,7 +74,7 @@ my (
 # Prepare test data depending on specific architecture/product
 sub prepare_test_data {
-    if (is_ppc64le || is_ppc64) {
+    if (is_pvm) {
```
I think you can reserve the previous condition. is_pvm check backend is 'spvm' or 'pvm_hmc' so that doesn't work .  And is_ppc64le will check ARCH=aarch64 works for all test. Create a new condition for test data if something different on powerVM or power 10 kvm.

lemon-suse commented:
The VR for power 10 kvm failed as incomplete, https://openqa.suse.de/tests/18424219#

hjluo commented:
> The VR for power 10 kvm failed as incomplete, [openqa.suse.de/tests/18424219](https://openqa.suse.de/tests/18424219#)

Updated to 119.1 build.

hjluo commented:
> > > The VR for power 10 kvm failed as incomplete, [openqa.suse.de/tests/18424219](https://openqa.suse.de/tests/18424219#)
> > 
> > 
> > Updated to 119.1 build.
> 
> Still have one incomplete job [openqa.suse.de/tests/18575962](https://openqa.suse.de/tests/18575962)

Rerun [passed](https://openqa.suse.de/tests/overview?arch=&flavor=&machine=&test=sles_raid0_unattended_dev&modules=&module_re=&group_glob=&not_group_glob=&comment=&distri=sle&version=16.0&build=119.1&groupid=318#)

hjluo commented:
>I think you can reserve the previous condition. is_pvm check backend is 'spvm' or 'pvm_hmc' so that doesn't work . And is_ppc64le will check ARCH=aarch64 works for all test. Create a new condition for test data if something different on powerVM or power 10 kvm.

this was already changed by following [PO's comments](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/22477#discussion_r2194168826).

rakoenig reviewed (approved):
Looks good.
